### DISCUSSION TITLE
Companion App not installed !

### DISCUSSION THREAD

**Discussion by Boyd365** [aclap-dev/video-downloadhelper#1137] [Help and Q&A]
### Your question

My Video DownloadHelper for Microsoft Edge quit working.
- Windows 10 x64 platform.
- The error I get is "Failed to start native messaging host."
- I uninstalled everything.
- Clean reboot.
- I downloaded newest versions of the extension and Companion App.
- Then, reinstalled the extension and Companion App.
- Same error.
- Rebooted ... still same error.
- Of course, it won't let me re-enter my license until the Companion App is working again. Argh!

Can anyone help me get this working again?

### Addon & CoApp version and details

version: 9.1.0.43
target: microsoft
channel: stable
lang: en-US
coapp: {"error":"Failed to start native messaging host.","found":false}
license: {"unset":true}
platform: x86-64 win
UA: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/127.0.0.0 Safari/537.36 Edg/127.0.0.0
{
    "used_history_button": true,
    "view_options": {
        "all_tabs": false,
        "hide_downloaded": false,
        "low_quality": false,
        "show_button_clean": true,
        "show_button_clean_all": false,
        "show_button_convert_local": false,
        "sort_by_status": true,
        "sort_reverse": false
    },
    "use_sidebar": true,
    "open_count_store": 2
}

**Comment by GrampaWildWilly** [comment]
I do not work for VDH.  I'm just another user.

My suspicion -- understand, I am guessing -- is that the uninstall of the CoApp didn't clean up some leftover in your registry.  I would use Revo Uninstall to uninstall the CoApp & try it all again.  I don't believe reboots should be necessary but I suppose they don't hurt.  You should search this forum for my references to `Revo` to get more detailed discussions.

**Reply by Boyd365** [reply]
- I uninstalled using `Revo`.
- Ran the 'Advanced' scan within `Revo` - it found two registry entries (I deleted them).
- Reinstalled CoApp.
- Same error.

**Comment by Mattmjs** [comment]
Do you close Edge browser first before installing companion app. If not you should do that and see if this resolves it.

**Reply by Boyd365** [reply]
Yes, I closed the Edge browser prior to installing CoApp.

**Comment by GrampaWildWilly** [comment]
The standard advice (which you can find in the Wiki) is to use the Windows Defender that comes built into Windows.  You should discard any other antivirus & firewall.  At the least, if you are using another antivirus and/or firewall, you're supposed to make every executable in the CoApp directory an exception in the antivirus & in the firewall.  Some users have reported that bits of the CoApp were quarantined.  This is, of course, a false positive.  My experience has been that Windows Defender has never flagged the CoApp.

Failing that, I'm out of ideas.  I don't use Edge, only Firefox.  Maybe somebody else has some ideas.

**Reply by Boyd365** [reply]
- I already use `Windows Defender`.
- I currently have nothing quarantined.
- Thank you.

Still looking for possible solutions from anyone.

**Comment by Boyd365** [comment]
I don't understand. CoApp quit working and refuses to re-install on my older (2019-era) laptop running Windows 10 - the same laptop that it has been running on very successfully for years!
This morning, I decided to install it on my newer (2024-era) laptop running Windows 11 and it works fine.
Too bad ... I really want to run this on the laptop that I normally leave in my study at home ... the 2019-era laptop.
So, I am still looking for possible solutions.

**Comment by Mattmjs** [comment]
There were a few issues previously for Edge but this was on Linux. I'm on Windows 10 and I added the extension to Edge.
It is able to find the companion app. Is your Edge browser the stable release & is the computer running as administrator ?
You've used the 64bit installer correct ? 

The documentaion for your error mesage says this: 

**Failed to start native messaging host.**

Check whether you have sufficient permissions to execute the native messaging host file.

**Reply by Boyd365** [reply]
- I had both the extension and CoApp installed on this computer for years.
- It worked FINE until last week (or at least that is when I noticed) when it just came up with the error of the CoApp not being installed.
- To the best of my knowledge, there was no major operating system update; nor did I install any new software.
- When I try to install (or recheck the installation), I receive the _**"Failed to start native messaging host."**_ error.
- I tried uninstalling ... even tried uninstalling using the `Revo Uninstall` program ... then reinstalling.
- I continually receive the error.
- Yes, the Edge browser in a stable version.
- Yes, the computer is running as administrator.
- Yes, I used the 64bit installer.
- Yes, I closed Edge prior to installing the CoApp.
- Yes, I use `Windows Defender` and nothing shows as being quarantined.
.
- I am not sure where the "native messaging host file" is in order to check whether I have sufficient permissions to execute it.
- But certainly, I must have had sufficient permission up until last week.

I am at a loss as to why it would work perfectly for years, then tell me CoApp was no longer installed (although it is), then get this native messaging host error!

**Reply by Mattmjs** [reply]
Where the companion app is installed is in
program files directory. It has 3 json files, one for each browser. These are the native messaging host files. See what permissions it shows for the Microsoft json. Use %programfiles% in windows search to get to the directory.

**Reply by Boyd365** [reply]
I checked ... Full Access is set.

**Reply by Mattmjs** [reply]
Did it look like this, if so I'm not sure what else it could be or where to go next.



![native messaging](https://github.com/user-attachments/assets/8b25cf2a-5c73-4983-8856-b3698c5dd16b) 
![native messaging_1](https://github.com/user-attachments/assets/0c445aa8-4c64-4bff-ae77-1710fc5903c9)

**Reply by Boyd365** [reply]
Yes.
I even tried changing the other three ("ALL APPLICATION PACKAGES", "ALL RESTRICTED APPLICATION PACKAGES", and "Users") to _**Full control**_ to no avail.
